agent/stats/engine.go

```diff
@@ -461,6 +464,15 @@ func (engine *DockerStatsEngine) GetInstanceMetrics(includeServiceConnectStats b
 			ContainerMetrics:      containerMetrics,
 		}
 
+		if includeServiceConnectStats {
+			serviceConnectStats, ok := engine.taskToServiceConnectStats[taskArn]
```

Comment:
Should we add the check to see if the returned service-connect stats has already been sent or not?

Reply:
Checking if service-connect stats has already been sent and setting it done once the SC stats are published needs to done in `tcs/client/client.go`. Will cover these changes in the next PR since `ServiceConnectStats` object is not accessible from tcs package currently

Reply:
Can we not add a warn log that there are "no new stats to be sent" from here and then set ServiceConnectWrapper for task metric to be nil ?

Reply:
Synced offline, will update this in new PR.